signal-desktop.sh

```diff
@@ -1,15 +1,66 @@
 #!/bin/bash
 
+report_warning() {
+    read -r -d '|' MESSAGE <<EOF
+Signal is being launched with the <b>plaintext password store</b> by
+default due to database corruption bugs when using the encrypted backends.
+This will leave your keys <b>unencrypted</b> on disk as it did in all previous versions.
+
+If you wish to experiment with the encrypted backend, set the environment variable
+<tt>SIGNAL_PASSWORD_STORE</tt> to <tt>gnome_libsecret</tt>, <tt>kwallet</tt>,
+<tt>kwallet5 or <tt>kwallet6</tt> depending on your desktop environment using
+Flatseal or the following command:
+
+<tt>flatpak override --env=SIGNAL_PASSWORD_STORE=gnome-libsecret org.signal.Signal</tt>
+
+Note that the encrypted backends are <b>experimental</b> and may cause data loss on some systems.
+
+Press <b>Yes</b> to proceed with <b>plaintext password store</b> or
+<b>No</b> to <b>exit</b>. |
+EOF
+    zenity --question --no-wrap --default-cancel --icon-name=dialog-warning --title "Warning" --text "$MESSAGE"
+
+    if [ "$?" -eq "1" ]; then
+        echo "Debug: Abort as user pressed no"
+        exit 1
+    else
+        touch "${XDG_CACHE_HOME}"/warning-shown
+    fi
+}
+
 EXTRA_ARGS=()
 
 declare -i SIGNAL_DISABLE_GPU="${SIGNAL_DISABLE_GPU:-0}"
 declare -i SIGNAL_DISABLE_GPU_SANDBOX="${SIGNAL_DISABLE_GPU_SANDBOX:-0}"
 
 # only kept for backward compatibility
-if (( ${SIGNAL_USE_WAYLAND:-0} )); then
+if ((${SIGNAL_USE_WAYLAND:-0})); then
     export ELECTRON_OZONE_PLATFORM_HINT="${ELECTRON_OZONE_PLATFORM_HINT:-auto}"
 fi
 
+declare -r SIGNAL_PASSWORD_STORE="${SIGNAL_PASSWORD_STORE:-basic}"
+
+case "${SIGNAL_PASSWORD_STORE}" in
+    basic | gnome-libsecret | kwallet | kwallet5 | kwallet6)
+        echo "Debug: Using password store: ${SIGNAL_PASSWORD_STORE}"
+        EXTRA_ARGS=(
+            "--password-store=${SIGNAL_PASSWORD_STORE}"
+        )
+        ;;
+    *)
+        echo "Error: SIGNAL_PASSWORD_STORE (${SIGNAL_PASSWORD_STORE}) must be one of the following: basic, gnome-libsecret, kwallet, kwallet5, kwallet6"
+        exit 1
+        ;;
+esac
+
+if [[ "${SIGNAL_PASSWORD_STORE}" == "basic" ]]; then
+    if [[ -f "${XDG_CACHE_HOME}"/warning-shown ]]; then
```

Comment:
This condition won't work, this will always fail for everyone as they haven't seen the warning yet and will always go to the else condition.

Reply:
You also have to take the password store out of the first if condition.

If someone switches from basic to encrypted and back to basic again the warning should be shown.

rm is pipes to true so it doesn't matter if the file previously didn't exist.